packages/openid4vc/src/openid4vc-issuer/router/federationEndpoint.ts

```diff
@@ -0,0 +1,102 @@
+import type { OpenId4VcIssuanceRequest } from './requestContext'
+import type { FederationKeyCallback } from '../../shared/federation'
+import type { Buffer } from '@credo-ts/core'
+import type { Router, Response } from 'express'
+
+import { getJwkFromKey } from '@credo-ts/core'
+import { createEntityConfiguration } from '@openid-federation/core'
+
+import { getRequestContext, sendErrorResponse } from '../../shared/router'
+import { OpenId4VcIssuerService } from '../OpenId4VcIssuerService'
+
+export interface OpenId4VcSiopFederationEndpointConfig {
+  /**
+   * The path at which the credential endpoint should be made available. Note that it will be
+   * hosted at a subpath to take into account multiple tenants and issuers.
+   *
+   * @default /.well-known/openid-federation
+   */
+  endpointPath: string
+
+  // TODO: Not sure about the property name yet.
+  //TODO: More information is needed than only the key also the client id etc
+  keyCallback: FederationKeyCallback<{
+    issuerId: string
+  }>
+}
+
+// TODO: It's also possible that the issuer and the verifier can have the same openid-federation endpoint. In that case we need to combine them.
+
+export function configureFederationEndpoint(router: Router, config: OpenId4VcSiopFederationEndpointConfig) {
+  router.get(config.endpointPath, async (request: OpenId4VcIssuanceRequest, response: Response, next) => {
+    const { agentContext, issuer } = getRequestContext(request)
+    const openId4VcIssuerService = agentContext.dependencyManager.resolve(OpenId4VcIssuerService)
+
+    try {
+      const issuerMetadata = openId4VcIssuerService.getIssuerMetadata(agentContext, issuer)
+      // TODO: Use a type here from sphreon
+      const transformedMetadata = {
```

Comment:
we can just use the type from our `shared/models/index` for issuer metadata here?